### PR TITLE
Kotlin cleanup + retrospective 

### PR DESCRIPTION
**Pull request by BradleyWire**
### Issues

With inspiration from the previous PR: https://github.com/wireapp/wire-android/pull/2599

I decided to have a small retro and look over the current codebase and get it up to scratch with the current coding style guidelines + more reusable ways of working. 
#### APK
[Download build #1220](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1220/artifact/build/artifact/wire-dev-PR2600-1220.apk)

### CODE REVIEW

File: app/src/main/kotlin/com/waz/zclient/settings/main/list/SettingsMainListItemsFactory.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import com.waz.zclient.R
+import com.waz.zclient.core.config.Config
+import com.waz.zclient.core.extension.stringArrayFromResource
+import com.waz.zclient.settings.main.model.SettingsMainItem
+
+class SettingsMainListItemsFactory private constructor() {
+
+    companion object {
+        fun generateList(context: Context): List<SettingsMainItem> {
+            val titles =
```
Comment (BradleyWire) [inline app/src/main/kotlin/com/waz/zclient/settings/main/list/SettingsMainListItemsFactory.kt]: Thanks @makingthematrix for this :)

File: app/src/main/kotlin/com/waz/zclient/settings/account/editphonenumber/UpdatePhoneDialogFragment.kt [hunk 2]
```diff
@@ -16,14 +16,13 @@ class UpdatePhoneDialogFragment : DialogFragment() {
...
-        arguments?.getString(CURRENT_PHONE_NUMBER_KEY, String.empty())
-            ?: String.empty()
+        arguments?.getString(CURRENT_PHONE_NUMBER_KEY, String.empty()) ?: String.empty()
     }
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog =
         AlertDialog.Builder(requireActivity())
             .setTitle(getString(R.string.pref__account_action__dialog__add_phone__confirm__title))
-            .setMessage(getString(R.string.pref__account_action__dialog__add_phone__confirm__message, phoneNumber))
+            .setMessage(getString(R.string.edit_phone_dialog_delete_phone_confirmation, phoneNumber))
```
Comment (mejdoo) [inline app/src/main/kotlin/com/waz/zclient/settings/account/editphonenumber/UpdatePhoneDialogFragment.kt]: You can take a look to all the string resources used by the Kotlin code in a separate PR

**Comment by mejdoo** [comment]
Great job! 👍

File: app/src/test/kotlin/com/waz/zclient/settings/account/SettingsAccountViewModelTest.kt [hunk 3]
```diff
@@ -167,7 +173,21 @@ class SettingsAccountViewModelTest : UnitTest() {
...
 
+    @Test
+    fun `given reset password is clicked, then url observer is notified`() {
+        `when`(accountsUrlConfig.url).thenReturn(TEST_ACCOUNT_CONFIG_URL)
+
+        viewModel.onResetPasswordClicked()
+
+
+        viewModel.resetPasswordUrlLiveData.observeOnce {
+            assert(it == "$TEST_ACCOUNT_CONFIG_URL$TEST_RESET_PASSWORLD_URL_SUFFIX")
```
Comment (gizemb) [inline app/src/test/kotlin/com/waz/zclient/settings/account/SettingsAccountViewModelTest.kt]: maybe we can use `shouldBe` for consistency.

**Pull request merged**